src/Server/src/Abstractions/IIdentityServerGroupService.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using IdOps.Model;
+
+namespace IdOps
+{
+    public interface IIdentityServerGroupService
+    {
+        Task<IReadOnlyList<IdentityServerGroup>> GetAllGroupsAsync(CancellationToken cancellationToken);
+        Task<IdentityServerGroup?> GetGroupByIdAsync(Guid id, CancellationToken cancellationToken);
+        Task<IdentityServerGroup?> GetGroupByTenantAsync(string tenant, CancellationToken cancellationToken);
+        Task<IEnumerable<IdentityServerGroup>> GetGroupsByUserTenants(CancellationToken cancellationToken);
+        Task<IdentityServerGroup> SaveAsync(SaveIdentityServerGroupRequest request, CancellationToken cancellationToken);
+    }
+
+    public record SaveIdentityServerGroupRequest(
```

Comment:
I guess this should go in a new file

Reply:
I kept the same style as in `IIdentityServerService`